internal/plugins/helm/api.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package helm
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold"
+
+	gencrd "github.com/operator-framework/operator-sdk/internal/generate/crd"
+	cmdutil "github.com/operator-framework/operator-sdk/internal/plugins/helm/internal"
+	sdkscaffold "github.com/operator-framework/operator-sdk/internal/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/helm"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/input"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+)
+
+type createAPIPlugin struct {
+	config *config.Config
+
+	// For help text.
+	commandName string
+
+	// Helm flags
+	//todo: Domain should be removed when the legacy layout be no longer supported
+	Domain           string
+	Group            string
+	Version          string
+	Kind             string
+	CRDVersion       string
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
```

Comment:
Rename HelmFoo to Foo for simplification? Looks like we are inside an struct which is meant to be used only inside the module.
Also pls add some comments to the fields? For example, ChartRef is a confusing field without a comment.